test/core/security/grpc_tls_crl_provider_test.cc

```diff
@@ -83,6 +101,110 @@ TEST(CrlProviderTest, StaticCrlProviderLookupBad) {
   EXPECT_EQ(crl, nullptr);
 }
 
+TEST(CrlProviderTest, DirectoryReloaderCrlLookupGood) {
+  auto result = experimental::CreateDirectoryReloaderProvider(
+      kCrlDirectory, std::chrono::seconds(1), nullptr);
+  ASSERT_TRUE(result.ok());
+  std::shared_ptr<CrlProvider> provider = std::move(*result);
+
+  CertificateInfoImpl cert = CertificateInfoImpl(kCrlIssuer);
+  auto crl = provider->GetCrl(cert);
+  ASSERT_NE(crl, nullptr);
+  ASSERT_EQ(crl->Issuer(), kCrlIssuer);
+
+  CertificateInfoImpl intermediate =
+      CertificateInfoImpl(kCrlIntermediateIssuer);
+  auto intermediate_crl = provider->GetCrl(intermediate);
+  ASSERT_NE(intermediate_crl, nullptr);
+  ASSERT_EQ(intermediate_crl->Issuer(), kCrlIntermediateIssuer);
+}
+
+TEST(CrlProviderTest, DirectoryReloaderCrlLookupBad) {
+  auto result = experimental::CreateDirectoryReloaderProvider(
+      kCrlDirectory, std::chrono::seconds(1), nullptr);
+  ASSERT_TRUE(result.ok());
+  std::shared_ptr<CrlProvider> provider = std::move(*result);
+
+  CertificateInfoImpl bad_cert = CertificateInfoImpl("BAD CERT");
+  auto crl = provider->GetCrl(bad_cert);
+  ASSERT_EQ(crl, nullptr);
+}
+
+std::string MakeTempDir() {
+  char templ[] = "/tmp/tmpdir.XXXXXX";
```

Comment:
micro-nit 1: s/templ/temp_dir, so that we have a slightly more descriptive variable name.

micro-nit 2: Can we make this a `std::string` instead of a `char[]`?

Reply:
I suggest making this a `static constexpr`, in which case it can't be `std::string`, as per the style guide:

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

It should be named `kTempDir`, as per:

https://google.github.io/styleguide/cppguide.html#Constant_Names

Reply:
`templ` here is a `template` for the temp directory name.
`mkdtemp` requires a non-const input

I've changed to `char directory_template[] = ...`, WDYT?

Reply:
I think Mark is suggesting to make a new constant at the top of the file:
```
static constexpr absl::string_view kTempDir = "/tmp/tmpdir.XXXXXX";
```
rather than recreating the local array each time we call the function.